llvm/lib/Target/AMDGPU/Disassembler/AMDGPUDisassembler.cpp

```diff
@@ -2366,8 +2366,12 @@ const MCExpr *AMDGPUDisassembler::createConstantSymbolExpr(StringRef Id,
                                                            int64_t Val) {
   MCContext &Ctx = getContext();
   MCSymbol *Sym = Ctx.getOrCreateSymbol(Id);
-  assert(!Sym->isVariable());
-  Sym->setVariableValue(MCConstantExpr::create(Val, Ctx));
+  int64_t Res = ~Val;
+  assert(!Sym->isVariable() ||
+         (Sym->getVariableValue()->evaluateAsAbsolute(Res) && Res == Val));
+  (void)Res;
```

Comment:
```suggestion
```

Reply:
If this is going to be updated, maybe we don't really want the assert at all, because generally there's no guarantee the user won't redefine the symbol and it wouldn't be nice to crash when they do.

Reply:
This makes me question whether implementing `UC_VERSION_*` as symbols _that the user can redefine_ was the best choice in the first place. But I don't have any alternative suggestion.

Reply:
> This makes me question whether implementing `UC_VERSION_*` as symbols _that the user can redefine_ was the best choice in the first place. But I don't have any alternative suggestion.

Doesn't look the best choice to me, but that's all we have at the moment.

Reply:
I have have replaced the assertion with a comment instead, so symbols always respected existing value even if it does not match.